app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt

```diff
@@ -1,18 +1,222 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.pvp.app.model.MealTask
+import com.pvp.app.model.SportActivity
+import com.pvp.app.model.SportTask
 import com.pvp.app.model.Task
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+// Parses Duration object to a string
+// Format <HH> h <mm> min
+fun getDurationString(duration: Duration): String {
+    val hours = duration.toHours()
+    val minutes = duration.minusHours(hours).toMinutes()
+
+    return buildString {
+        if (hours > 0) append("$hours h")
+        if (minutes > 0) append("${if (hours > 0) " " else ""}$minutes min")
+    }
+}
+@Composable
+fun MealTaskBoxBody(
+    mealTask: MealTask
+){
+
+    Text(
+        "Main ingredient: " + mealTask.recipe,
+        textAlign = TextAlign.Left,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 8.dp)
+    )
+
+    if(mealTask.duration != null){
+        Text(
+            "Duration: " + getDurationString(mealTask.duration!!),
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
+@Composable
+fun SportTaskBoxBody(
+    sportTask: SportTask
+) {
+
+    sportTask.activity?.let { activity ->
+        val activityText = buildString {
+            append(activity.title)
+            when {
+                sportTask.distance != null ->
+                    append(" for ${sportTask.distance} km")
+
+                sportTask.duration != null ->
+                    append(" for ${getDurationString(sportTask.duration!!)}")
+            }
+        }
+
+        Text(
+            text = activityText,
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
 
+@Composable
+fun TaskBoxBody(
+    task: Task
+) {
+
+    task.description?.let { description ->
+        Text(
+            text = description,
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+
+    task.duration?.let { duration ->
+        Text(
+            text = "Duration: ${getDurationString(duration)}",
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
 @Composable
 fun TaskBox(
     task: Task
 ) {
 
+    // Later task.isCompleted should be used
+    var checked by remember {
+        mutableStateOf(false)
+    }
+
+    Card(
+        shape = RectangleShape,
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
+    ) {
+
+        Column(
+        ) {
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
```

Comment:
Single parameter within brackets _could_ stay in a single line. Can leave it as it is, if wanted.